addons/web/static/src/js/views/form_widgets.js

```diff
@@ -888,6 +888,23 @@ var FieldSelection = common.AbstractField.extend(common.ReinitializeFieldMixin,
     }
 });
 
+var TimeZone = FieldSelection.extend({
+    initialize_content: function() {
+        var self = this;
+        return self.alive(new Model('res.users').call('read', [[session.uid], ['tz_offset']])).then(function(result) {
```

Comment:
why not creating separate method check_timezone (may be util method), put common stuff there as you are doing same stuff here and in user preferences